### PR TITLE
Fix invalid date selection issue for e2e_tests

### PR DESCRIPTION
**Pull request by mkrump**
## Description

The cypress test is selecting a day from the date picker for the current month. This will fail when the date is a weekend or holiday. Assuming that it's not possible to hard code the date and enter the date and that have to interact via the date picker (as I think that would probably be the easiest way to resolve)? 

## Reviewer Notes

Wasn't sure the best way to resolve this. Also, not really sure what's been tried here, so if you all could weigh in that would be appreciated. Two commits with two different approaches:
* https://github.com/transcom/mymove/compare/34b5ab1...e9188bc3109009fb1db986f11f9bcb31ea737b9f make a call to the `available_move_dates` endpoint and uses the result. Maybe overly complicated for what we're trying to do here, but uses our holiday calendar / don't have to maintain another calendar. 
* https://github.com/transcom/mymove/compare/34b5ab1...affb732cae516bded3e13daa2d080035d8362aa2 uses `moment-business-days`. This is simpler, but requires that you add a holiday calendar, that we'd have to update going forward, whereas that's already being done in the `available_move_dates` endpoint. 

*** Updated ***
Was directed to this [story](https://www.pivotaltracker.com/n/projects/2136865/stories/166531458). This specific test was failing due to the shipments model validation rules. Adding a third option (remove the actual delivery date `DateIsWorkday` validation rule), but didn't have the context to know if that was correct to remove it entirely.

* https://github.com/transcom/mymove/compare/34b5ab1...23087f297feae34fc23a6dc1db67e2f1cebec0dc remove shipment model validation

## Setup

```sh
make e2e_test
```

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/n/projects/2226219/stories/167026270) for this change

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
I prefer the use of the `available_move_dates` endpoint because it keeps our maintenance burden low for the holiday stuff.

**Comment by lynzt** [comment]
If we already have a way to calculate holidays, I'd use that... As Matt said, prevents us from having to maintain another holiday calendar. 
However, w/ the update of the business rules seems like we could get rid of testing for weekend dates all together since the TSP can choose a weekend.
Question: Are TSPs allowed to choose a date that might fall on a holiday? Do we still need a check for that?

**Comment by tinyels** [comment]
I'm of two minds about this:
The pragmatic mind says that we are moth-balling HHGs and should do the least amount of work possible to get this test passing. That is, we could remove the verification entirely.
The perfectionist says that we want to check for weekends and holidays if the request is from a service member. It also say that the business rules for pickup and packing date should be corrected too.

For this story, I prefer the `available_move_dates`, since a) you have already done the work and b) it will be resilient to any changes we make to that business logic going forward.

File: Dockerfile.cypress [hunk 1]
```diff
@@ -4,7 +4,7 @@ FROM cypress/base:10
 # We use `npm install` here because you can't selectively `yarn install` packages and we don't care about a lockfile and
 # `yarn add` won't read from our existing package.json
 COPY package.json /package.json
-RUN npm install --save-dev mocha mocha-multi-reporters mocha-junit-reporter
```
Comment (mkrump) [inline Dockerfile.cypress]: had to add `moment` to cypress container since using it within cypress now.

File: Dockerfile.cypress [hunk 2]
```diff
@@ -4,7 +4,7 @@ FROM cypress/base:10
...
+RUN npm install --save-dev mocha mocha-multi-reporters mocha-junit-reporter moment
```
Comment (mkrump) [inline Dockerfile.cypress]: had to add `moment` to the cypress container since using it within cypress now.

**Comment by mkrump** [comment]
thanks for the feedback everyone. seems like the consensus is to give the `available_move_dates` approach a shot. updated PR to use that.

**Pull request merged**